tasks/install.yml

```
@@ -63,7 +63,20 @@
     dest: "{{ promtail_tmp_dir }}/{{ promtail_version }}_promtail-linux-{{ go_arch }}.zip"
     force: True
     checksum: "{{ promtail_custom_checksum if promtail_custom_checksum else 'sha256:' + __promtail_checksum }}"
-  when: not promtail_binary.stat.exists
+  when:
+    - not promtail_binary.stat.exists
+    - promtail_binary_local_dir | length == 0
+
+- name: propagate locally distributed promtail archive
```

nitpick from the linter

> ERROR: [ANSIBLE0007] Standard 'Name of tasks and handlers must be formatted' not met:
> ... tasks/install.yml:70: name 'propagate locally distributed promtail archive' should start with uppercase

```suggestion
- name: Propagate locally distributed promtail archive
```